### ISSUE TITLE
unneeded_parentheses_in_closure_argument doesn't work with capture lists

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1817]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.22.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Paste your configuration file:

```yml
opt_in_rules:
  - unneeded_parentheses_in_closure_argument
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. Nop
* Which Xcode version are you using (check `xcode-select -p`)? Xcode 9 
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This should trigger unneeded_parentheses_in_closure_argument
foo.bar { [weak self] (x,y) in }
```
